actioncable/test/channel/stream_test.rb

```diff
@@ -94,6 +94,33 @@ class StreamTest < ActionCable::TestCase
       end
     end
 
+    test "success_callback is invoked" do
+      run_in_eventmachine do
+        connection = TestConnection.new
+
+        channel = OrderedCallbackChannel.new connection, "{id: 9}", id: 9
+
+        fake_pubsub = Class.new do
+          def subscribe(channel_name, handler, success)
+            success.call
+          end
+        end.new
+
+        connection.stub(:pubsub, fake_pubsub) do
+          channel.subscribe_to_channel
+          wait_for_async
+        end
+
+        assert_equal [
+          :subscribed_method_end,
+          :after_subscribe_callback,
+          :room_1_success,
+          :room_2_success,
+          :transmit_subscription_confirmation
```

Comment:
This test surfaces the `after_subscribe` callback order happening _before_ subscription confirmation. It doesn't imply that's intended though (#25333). I'm planning to look at that next.